driver/src/db_interface.rs

```diff
@@ -0,0 +1,153 @@
+extern crate rustc_hex;
+extern crate web3;
+
+use crate::models;
+
+use mongodb::bson;
+use mongodb::db::ThreadedDatabase;
+use mongodb::{Client, ThreadedClient};
+
+use web3::types::H256;
+
+use std::io;
+use std::io::{Error, ErrorKind};
+
+pub trait DbInterface {
+    fn get_current_balances(
+        &self,
+        current_state_root: H256,
+    ) -> Result<models::State, Error>;
+    fn get_deposits_of_slot(
+        &self,
+        slot: i32,
+    ) -> Result<Vec<models::Deposits>, io::Error>;
+}
+
+#[derive(Clone)]
+pub struct DbMongoInstance {
```

Comment:
nit: naming, a struct is not really an instance (an instance of a struct is). Maybe just call this MongoDB?